lib/pure/json.nim

```diff
@@ -446,7 +446,8 @@ proc hash*(n: JsonNode): Hash =
 
 proc hash*(n: OrderedTable[string, JsonNode]): Hash =
   for key, val in n:
-    result = result xor (hash(key) !& hash(val))
+    result = result !& hash(key)
+    result = result !& hash(val)
```

Comment:
Looks like a regression to me, for JSON the order must not matter.

Reply:
indeed; there's a bit of a disconnect between OrderedTable and the fact that `==` for json ignores order; I'm now using `hashUnordered` as well for JObject case (still more robust than xor for same reasons as explained earlier); see latest commits